artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/AbstractJournalStorageManager.java

```diff
@@ -1244,6 +1244,9 @@ private static PageSubscription locateSubscription(final long queueID,
          if (queueInfo != null) {
             SimpleString address = queueInfo.getAddress();
             PagingStore store = pagingManager.getPageStore(address);
```

Comment:
private Map<SimpleString, Collection<Integer>> getPageInformationForSync(PagingManager pagingManager) throws Exception {
      Map<SimpleString, Collection<Integer>> info = new HashMap<>();
      for (SimpleString storeName : pagingManager.getStoreNames()) {
         PagingStore store = pagingManager.getPageStore(storeName);
         info.put(storeName, store.getCurrentIds());
         store.forceAnotherPage();
      }
      return info;
   }

In getPageInformationForSync(), do we need to judge null PagingStore?

Reply:
@wy96f I'm not following, that method is used purely for the File journal during replication.  The null check on paging store is due to, the additional bits in getPageStore(SimpleString storeName).

```java
      if (managementAddress != null && storeName.startsWith(managementAddress)) {
         return null;
      }
```

Reply:
@mtaylor I'm considering to be compatible with old server. There maybe some old management address paging folders. When we upgrade,  getPageStore() may return null. That doesn't matter as we can delete old management folders and restart.

Reply:
Might be worth adding some logic to clean this up automatically. Otherwise lots of users probably wont know or might miss it if its a release note.